artifact/version/version.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef MENDER_ARTIFACT_VERSION_PARSER_HPP
+#define MENDER_ARTIFACT_VERSION_PARSER_HPP
+
+#include <string>
+
+#include <common/expected.hpp>
+#include <common/error.hpp>
+#include <common/io.hpp>
+
+namespace artifact {
+namespace v3 {
+namespace version {
+
+using namespace std;
+
+namespace io = mender::common::io;
+namespace expected = mender::common::expected;
+namespace error = mender::common::error;
+
+
+enum ErrorCode {
+	NoError = 0,
+	ParseError,
+	VersionError,
+	FormatError,
```

Comment:
I have a question about this, but it's an abstract question, hard to put into words here. Ping me when you have time.